rpcserver.go

```diff
@@ -1747,7 +1747,7 @@ func (r *rpcServer) DisconnectPeer(ctx context.Context,
 	// In order to avoid erroneously disconnecting from a peer that we have
 	// an active channel with, if we have any channels active with this
 	// peer, then we'll disallow disconnecting from them.
-	if len(nodeChannels) > 0 && !r.cfg.UnsafeDisconnect {
+	if len(nodeChannels) > 0 {
```

Comment:
I think this commit changes the behavior of `lnd`. Before, because of the default value being `true`, we would never get this error in the default configuration. Now we cannot disconnect active channels, which might be an issue if the channel gets stuck.

Reply:
Indeed. Tho precisely speaking, it's not default to `true`, but only a constant value of `true` because we set it here
https://github.com/lightningnetwork/lnd/blob/cf4f468ab5c70e644d8974145d1ffd74a3dabc35/config.go#L579
And deprecated this flag a while back, so it was not read from user's config file. This means we can always disconnect whether there are active channels or not, which, I think was not the intention that we removed that config in the first place, as it would always skip this check. Meanwhile, yeah, think we should forbid disconnection when there are open channels, if a user really insists in disconnecting, the only option is to shut down the node, hopefully that won't be tried.

Reply:
I'm not sure if we can just change the behavior here. I mean, I get why it's not a good idea to disconnect if there are active channels. But isn't that today the only way you can fix an inactive channel due to connection bugs? And aren't there tools that explicitly disconnect/re-connect your channels periodically to make sure they remain active (thinking of LNDg and similar tools)?
So if you have two channels with a node and one is inactive and one is active, how would you be able to re-connect to force the inactive channel to re-establish?

Sure, the underlying issue is that channels become inactive (while the p2p connection itself is healthy) in the first place, but I'm not sure we have properly found and fixed all those occurrences?

Reply:
Maybe we should just add a new RPC that does this reconnection😵‍💫
>  So if you have two channels with a node and one is inactive and one is active, how would you be able to re-connect to force the inactive channel to re-establish?

I think this is precisely the issue, that we should have a channel-level RPC to kick off the reestablishment, if it fixes the problem at all. Future PRs tho. So what's your suggestion here? I have two options in mind,
1. remove the line `if len(nodeChannels) > 0` since we allow disconnection happening anyway, or,
2. make `GetUnsafeDisconnect` to return `true`, in case we want to enforce the active channel checks in the future, we can then change it to be `false`.

Reply:
I would personally favor option 2, but curious to hear what other people think.

Reply:
And yes, I think a channel-level re-establishment RPC does sound like a good idea.

Reply:
> Tho precisely speaking, it's not default to true, but only a constant value of true because we set it here

Setting that constant there does make the default true though?

> , I think was not the intention that we removed that config 

yeah you are probably right!

> Maybe we should just add a new RPC that does this reconnection

Yeah this seems important to have as an option for users 👍 

agreed that we should not change the default behaviour until that RPC option is added

Reply:
Cool went with option 2